bin/instruments/qmi_tenma_72psu.py

```diff
@@ -0,0 +1,150 @@
+#! /usr/bin/env python3
+
+"""Command line client for the Tenma 72-series power supplies."""
+
+import argparse
+from contextlib import nullcontext, AbstractContextManager
+import sys
+
+import qmi
+from qmi.instruments.tenma import *  # This may look like unused in IDE but will import all PSU classes.
+from qmi.utils.context_managers import start_stop, open_close
+
+
+def main() -> int:
+
+    parser = argparse.ArgumentParser()
+    parser.description = "Command line client for the Tenma 72 series Power Supply Units."
+
+    parser.add_argument("--model", type=str, help="The model number of the Tenma 72 series PSU (e.g. 2550)")
+    parser.add_argument("--channel", type=int, help="Channel or memory slot of instrument", const=None)
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument("--udp", type=str, help="IP address of device.")
+    group.add_argument("--serial", type=str, help="Serial address of device (COMx, /dev/ttySx).")
+
+    parser.add_argument("--idn", action="store_true", help="Get instrument identification")
+    parser.add_argument("--status", action="store_true", help="Get instrument status")
+    # setters/getters
+    parser.add_argument("--current", type=str, help="Get/Set current in Amperes", nargs='?', const='get',)
+    parser.add_argument("--voltage", type=str, help="Get/Set voltage in Volts", nargs='?', const='get',)
+    parser.add_argument("--output", action="store_true", help="Enable output", default=None)
+    parser.add_argument("--no-output", dest="output", action="store_false", help="Disable output")
+    parser.add_argument("--ip", type=str, help="Get/Set IP address", nargs='?', const='get')
+    parser.add_argument("--dhcp", type=str, help="Get/Set IP address", nargs='?', const='get')
+    parser.add_argument("--port", type=str, help="Get/Set IP port number", nargs='?', const='get')
+    parser.add_argument("--mask", type=str, help="Get/Set IP subnet mask", nargs='?', const='get')
+    parser.add_argument("--gateway", type=str, help="Get/Set IP gateway", nargs='?', const='get')
+    args = parser.parse_args()
+
+    instr: globals()[f"Tenma72_{args.model}"]
```

Comment:
i'm not sure how globals works, but I would try with `instr = getattr(qmi.instrument.tenma, f"Tenma_{args.model}")`

Reply:
Both work, and give the correct instance back. But I suppose the `getattr` version is safer as it limits the search in `qmi.instruments.tenma` package, and not ALL globals. Even though I really doubt there would be some other global variable called `Tenma_xxx`...